frontend/src/components/AuthManager.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, FunctionComponent } from 'react';
+import { GoogleLogin } from 'react-google-login';
+import { useHistory, useLocation } from "react-router-dom";
+import useClientId from '../hooks/useClientId';
+import AuthContext from '../context/auth';
+import LandingPage from './LandingPage';
+
+export const AuthManager: FunctionComponent = ({ children }) => {
+  const [signedIn, setSignedIn] = useState(false);
+  const clientId = useClientId();
+  let history = useHistory();
+  const { pathname } = useLocation();
+
+  function logout() {
+    localStorage.clear();
+    setSignedIn(false);
+    history.push('/');
+  }
+
+  async function onSignIn(googleUser: any) {
+    let token = googleUser.getAuthResponse().id_token;
+    let email = googleUser.profileObj.email;
```

Comment:
again using const here, and i think destructuring would look cleaner
```
const { id_token: token } = googleUser.getAuthResponse();
const { email } = googleUser.profileObj;
// or even const { profileObj: { email } } = googleUser;
```

Reply:
Nice suggestion. Should have cleaned up more, but a lot of this was copy-pasted from Lucy's original component.